### ISSUE TITLE
AoE modal doesn't close on `esc` key

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#4072]
## Area or Component
The Ask on Entry modal in the test card

## Quick Summary
The modal doesn't respond to the `esc` key.

### How to Replicate
1. Start a new test
2. Start filling out the questionnare
3. Try to ESC out of the modal
4. Nothin happens

## Accessibility Violation
ESC key does not close modal: https://cdc-devtools.dequecloud.com/rule-help/en/modal-no-esc

## Action Requested
Change the modal to close on `esc`. See #4028 for more details about how to implement this change.

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=981460209&range=B34:T34
